### ISSUE TITLE
The ability to deploy version 5.0

### ISSUE THREAD

**Issue by redl1k** [mitre/caldera#2996]
Hi everyone. Encountered a problem with deploying “Caldera” version 5.0. After reading other Issue to this project and realized that version 5.0 has a number of issues that do not allow to deploy this version.

Tried to deploy version 4.2, there were no problems with it and everything works. 

And a question for the community. Are there any fixes for the problems that will help deploy version 5.0? Or should I wait for a official bug fix?

One of the problems with deploying “Caldera v5.0” is running a python server that cannot start and crashes with the following error... attached log 
```
$>python3 server.py --build
2024-06-05 09:49:46 INFO     Using main config from conf/local.yml                                                                                                                               server.py:236
2024-06-05 09:49:47 INFO     Setting VueJS environment file.                                                                                                                                     server.py:151
                    INFO     Building VueJS front-end.                                                                                                                                           server.py:274

up to date, audited 770 packages in 2s

100 packages are looking for funding
  run `npm fund` for details

11 vulnerabilities (7 moderate, 3 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> magma@0.0.0 build
> node prebundle.js && vite build

Copying all plugin GUI source files to magma
Copying over "access" files...
Copying over "atomic" files...
Copying over "builder" files...
Copying over "compass" files...
Copying over "debrief" files...
Copying over "emu" files...
Copying over "gameboard" files...
Copying over "human" files...
Copying over "manx" files...
Copying over "response" files...
Copying over "sandcat" files...
Copying over "ssl" files...
Copying over "stockpile" files...
Copying over "stockpile11111" files...
Copying over "training" files...
Plugin GUI source files copied!
vite v2.9.15 building for production...
✓ 0 modules transformed.
[vite:vue] Failed to resolve vue/compiler-sfc.
@vitejs/plugin-vue requires vue (>=3.2.25) to be present in the dependency tree.
error during build:
Error: Failed to resolve vue/compiler-sfc.
@vitejs/plugin-vue requires vue (>=3.2.25) to be present in the dependency tree.
    at resolveCompiler (/home/user/Desktop/caldera/plugins/magma/node_modules/@vitejs/plugin-vue/dist/index.js:3955:11)
    at Object.buildStart (/home/user/Desktop/caldera/plugins/magma/node_modules/@vitejs/plugin-vue/dist/index.js:4686:46)
    at /home/user/Desktop/caldera/plugins/magma/node_modules/rollup/dist/shared/rollup.js:22834:37
    at async Promise.all (index 4)
    at async /home/user/Desktop/caldera/plugins/magma/node_modules/rollup/dist/shared/rollup.js:23622:13
    at async catchUnfinishedHookActions (/home/user/Desktop/caldera/plugins/magma/node_modules/rollup/dist/shared/rollup.js:23132:20)
    at async rollupInternal (/home/user/Desktop/caldera/plugins/magma/node_modules/rollup/dist/shared/rollup.js:23620:5)
    at async doBuild (/home/user/Desktop/caldera/plugins/magma/node_modules/vite/dist/node/chunks/dep-689425f3.js:41710:24)
    at async build (/home/user/Desktop/caldera/plugins/magma/node_modules/vite/dist/node/chunks/dep-689425f3.js:41552:16)
    at async CAC.<anonymous> (/home/user/Desktop/caldera/plugins/magma/node_modules/vite/dist/node/cli.js:738:9)
npm ERR! code 1
npm ERR! path /home/user/Desktop/caldera/plugins/magma
npm ERR! command failed
npm ERR! command sh -c node prebundle.js && vite build

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2024-06-05T13_49_50_512Z-debug.log
Traceback (most recent call last):
  File "/home/user/Desktop/caldera/server.py", line 276, in <module>
    subprocess.run(["npm", "run", "build"], cwd=MAGMA_PATH, check=True)
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 1.
```

**Comment by redl1k** [comment]
Strange, I updated nodejs but the system didn't want to accept it....... I had to completely reinstall NodeJS to the correct version and the script to deploy ‘Caldera’ ran, thanks for the tip :)

But at the moment I'm facing the same problem as in the previous release. Namely, it is impossible to log into the platform. And at this point I guess we need to wait for an official bug fix?

**Comment by elegantmoose** [comment]
We thought we fixed it. On the queue to review again. Apologies the inconvenience.

**Comment by jbaptperez** [comment]
@redl1k, You can apply the workaround detailed [here](https://github.com/mitre/caldera/issues/2984#issuecomment-2115959067), expecting a fix (issue #2984).

**Comment by redl1k** [comment]
@jbaptperez Thanks for the reply. the solution helped :)
